scipy/spatial/src/distance_impl.h

```diff
@@ -167,26 +167,27 @@ yule_bool_distance_char(const char *u, const char *v, npy_intp n)
     npy_intp ntt = 0, nff = 0, nft = 0, ntf = 0;
 
     for (i = 0; i < n; i++) {
-        ntt += (u[i] && v[i]);
-        ntf += (u[i] && !v[i]);
-        nft += (!u[i] && v[i]);
-        nff += (!u[i] && !v[i]);
+        npy_bool x = u[i], y = v[i];
+        ntt += x & y;
+        ntf += x & (!y);
+        nft += (!x) & y;
```

Comment:
the problem is it is not guaranteed the lowerbit is set for a true boolean, in numpy a true boolean is determined by b != 0, b &1 would be false if b is for example 2

its an edge case but we had bugs about that in numpy
npy_bool will not help as that maps to char not c99 bool (where & would indeed be correct)